components/Wizard/formComponents/Packages.js

```diff
@@ -0,0 +1,537 @@
+  SearchInput,
+  Spinner,
+  TextContent,
+} from "@patternfly/react-core";
+import { AngleDoubleLeftIcon, AngleLeftIcon, AngleDoubleRightIcon, AngleRightIcon } from "@patternfly/react-icons";
+import { connect } from "react-redux";
+import api from "../../../core/api";
+import * as composer from "../../../core/composer";
+import { updateBlueprintComponents } from "../../../core/actions/blueprints";
+
+// the fields isHidden and isSelected should not be included in the package list sent for image creation
+const removePackagesDisplayFields = (packages) =>
+  packages.map((pack) => ({
+    name: pack.name,
+    summary: pack.summary,
+  }));
+
+const Packages = ({ defaultArch, ...props }) => {
+  const { change, getState } = useFormApi();
+  const { input } = useFieldApi(props);
+  const [packagesSearchName, setPackagesSearchName] = useState(undefined);
+  const [filterAvailable, setFilterAvailable] = useState(undefined);
+  const [filterChosen, setFilterChosen] = useState(undefined);
+  const [packagesAvailable, setPackagesAvailable] = useState([]);
+  const [packagesAvailableFound, setPackagesAvailableFound] = useState(true);
+  const [packagesAvailableLoading, setPackagesAvailableLoading] = useState(false);
+  const [packagesChosen, setPackagesChosen] = useState([]);
+  const [packagesChosenFound, setPackagesChosenFound] = useState(true);
+  const [packagesChosenLoading, setPackagesChosenLoading] = useState(false);
+  const [focus, setFocus] = useState("");
+  const [totalPackagesAvailable, setTotalPackagesAvailable] = useState(undefined);
+  const [packagesAvailablePage, setPackagesAvailablePage] = useState(1);
+  const [packagesAvailablePerPage, setPackagesAvailablePerPage] = useState(100);
+  const [additionalPackagesAvailableLoading, setAdditionalPackagesAvailableLoading] = useState(false);
+
+  // this effect only triggers on mount
+  useEffect(() => {
+    const fetchPackagesChosen = async () => {
+      setPackagesChosenLoading(true);
+      const result = await composer.depsolveBlueprint(props.blueprintName);
+      const blueprint = result.blueprints[0].blueprint;
+      const blueprintPackages = blueprint.packages.concat(blueprint.modules);
+      updateBlueprintComponents(blueprintPackages);
+      const packageNames = blueprintPackages.map((pkg) => pkg.name);
+      if (packageNames.length !== 0) {
+        const packageInfo = await composer.getComponentInfo(packageNames);
+        const selectedPackages = packageInfo.map((pkg) => ({ name: pkg.name, summary: pkg.summary }));
+        setPackagesChosenSorted(selectedPackages);
+        change(input.name, removePackagesDisplayFields(selectedPackages));
+      }
+      setPackagesChosenLoading(false);
+    };
+
+    const formPackages = getState()?.values?.["selected-packages"];
+
+    if (formPackages) {
+      setPackagesChosenSorted(formPackages);
+    } else {
+      fetchPackagesChosen();
+    }
+  }, []);
+
+  const updateBlueprintComponents = (packageList) => {
```

Comment:
Can we move the actual blueprint update to the final step? I would like the actual blueprint update to be on confirmation i.e. Create Image button or a Save Bluepoint button

Reply:
Great idea. The current behavior more closely resembles the existing cockpit-composer behavior, where adding or removing packages on the `blueprintEdit` page modifies the workspace. I would actually prefer it to behave more like Image Builder Front End, which would be the result of implementing your suggestion.